api/types.go

```diff
@@ -141,6 +141,12 @@ type ListResponse[T any] struct {
 	LastUpdateIndex `json:"-"`
 }
 
+func (r ListResponse[T]) SetHeaders(h http.Header) {
+	if r.LastUpdateIndex.Index > 0 {
+		h.Set("Last-Update-Index", strconv.FormatInt(r.LastUpdateIndex.Index, 10))
+	}
+}
```

Comment:
I forgot to comment on this. I moved this over from `ListGrantsResponse` because we need this header set for previous versions of the `ListGrants` endpoint as well.

I think #3993 would allow us to move this to the `BlockingRequest` type, that we can embed in the list responses.